### DISCUSSION TITLE
composer require spatie/laravel-export fails on Laravel 8 due to Guzzle

### DISCUSSION THREAD

**Discussion by mirkin** [spatie/laravel-export#75] [Q&A]
This only seems to happen with the latest version of Laravel it works with the previous version.

Problem 1
    - Can only install one of: guzzlehttp/guzzle[6.5.x-dev, 7.0.1].
    - Can only install one of: guzzlehttp/guzzle[7.0.1, 6.5.x-dev].
    - Can only install one of: guzzlehttp/guzzle[6.5.x-dev, 7.0.1].
    - spatie/laravel-export 0.3.4 requires guzzlehttp/guzzle ^6.3 -> satisfiable by guzzlehttp/guzzle[6.5.x-dev].
    - Installation request for spatie/laravel-export ^0.3.4 -> satisfiable by spatie/laravel-export[0.3.4].
    - Installation request for guzzlehttp/guzzle (locked at 7.0.1, required as ^7.0.1) -> satisfiable by guzzlehttp/guzzle[7.0.1].

**Comment by runthis** [comment]
This is still an issue

`  Problem 1
    - Installation request for spatie/laravel-export ^0.3.9 -> satisfiable by spatie/laravel-export[0.3.9].
    - Conclusion: remove guzzlehttp/psr7 2.0.0
    - Conclusion: don't install guzzlehttp/psr7 2.0.0
    - spatie/laravel-export 0.3.9 requires spatie/crawler ^4.7 -> satisfiable by spatie/crawler[4.7.0, 4.7.1, 4.7.2, 4.7.3, 4.7.4, 4.7.5, 4.7.6, v4.x-dev].
    - spatie/crawler 4.7.0 requires guzzlehttp/psr7 ^1.4 -> satisfiable by guzzlehttp/psr7[1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 1.x-dev].
    - spatie/crawler 4.7.1 requires guzzlehttp/psr7 ^1.4 -> satisfiable by guzzlehttp/psr7[1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 1.x-dev].
    - spatie/crawler 4.7.2 requires guzzlehttp/psr7 ^1.4 -> satisfiable by guzzlehttp/psr7[1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 1.x-dev].
    - spatie/crawler 4.7.3 requires guzzlehttp/psr7 ^1.4 -> satisfiable by guzzlehttp/psr7[1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 1.x-dev].
    - spatie/crawler 4.7.4 requires guzzlehttp/psr7 ^1.4 -> satisfiable by guzzlehttp/psr7[1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 1.x-dev].
    - spatie/crawler 4.7.5 requires guzzlehttp/psr7 ^1.4 -> satisfiable by guzzlehttp/psr7[1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 1.x-dev].
    - spatie/crawler 4.7.6 requires guzzlehttp/psr7 ^1.4 -> satisfiable by guzzlehttp/psr7[1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 1.x-dev].
    - spatie/crawler v4.x-dev requires guzzlehttp/psr7 ^1.4 -> satisfiable by guzzlehttp/psr7[1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.5.2, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 1.x-dev].
    - Can only install one of: guzzlehttp/psr7[1.7.0, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.8.0, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.8.1, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.8.2, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.x-dev, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.4.0, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.4.1, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.4.2, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.5.0, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.5.1, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.5.2, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.6.0, 2.0.0].
    - Can only install one of: guzzlehttp/psr7[1.6.1, 2.0.0].
    - Installation request for guzzlehttp/psr7 (locked at 2.0.0) -> satisfiable by guzzlehttp/psr7[2.0.0].


Installation failed, reverting ./composer.json to its original content.`

**Comment by stosveta** [comment]
Problem 1
    - spatie/crawler[4.7.6, ..., v4.x-dev] require guzzlehttp/psr7 ^1.4 -> found guzzlehttp/psr7[1.4.0, ..., 1.x-dev] but the package is fixed to 2.1.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - spatie/crawler[4.7.0, ..., 4.7.5] require php ^7.1 -> your php version (8.0.10) does not satisfy that requirement.
    - spatie/laravel-export 0.3.9 requires spatie/crawler ^4.7 -> satisfiable by spatie/crawler[4.7.0, ..., v4.x-dev].
    - Root composer.json requires spatie/laravel-export ^0.3.9 -> satisfiable by spatie/laravel-export[0.3.9].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
